R/Microclusters.R

```diff
@@ -420,15 +420,15 @@ poolMatrixCols_Inner <- function(expr, pools) {
 
 
 #' Performs a binary search on a depth d such that 
-#' if depth(u, v) <= d then u and v are in the same cluster
+#' if depth(LCA(u, v)) <= d then u and v are in the same cluster
 #'
 #' @param tree object of class phylo
 #' @param reach number of clusters to attempt to generate
 #' @return List of clusters, each entry being a vector of indices representing
 #' samples in the cluster.
 #' 
 #' @export
-treeCluster1 <- function(tree, reach=10) {
+depthBasedTreeCluster <- function(tree, reach=10) {
```

Comment:
This function name is a lot more clear! But I don't like the argument `reach` -- let's use something more informative.

Reply:
How about target?

Reply:
No, I think `depth` would be more reasonable.

We should have informative names for the other clustering algorithms too.

Reply:
What about numClusters?

Reply:
No - because this parameter does not control the number of clusters, but rather the depth at which you cut. `numClusters` would be unclear to a user.

Reply:
The depth isn't the parameter though, we're doing a binary search for the depth to yield the specified number of clusters.